### PR TITLE
feat: add FluxRecord.row with response data stored in Array

### PR DESCRIPTION
**Pull request by michaelahojna**
## Proposed Changes

Adding possibility of accessing response data in Array FluxRecord.row.

In case of using pivot on data, where field contains labels that occur by default in the annotated CSV (f.e. "result" or "table"), could be duplicated column names in response. In that case FluxRecord.values, which hold only unique keys, doesn't show complete data. This edge case is solved by using FluxRecord.row.

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] `make test` completes successfully
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

File: examples/RecordRowExample.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    "precision" => InfluxDB2\Model\WritePrecision::S
+]);
+
+//
+// Write test data into InfluxDB
+//
+$writeApi = $client->createWriteApi();
+
+foreach (range(1, 5) as $i) {
+    $writeApi->write("point,table=my-table result={$i}", InfluxDB2\Model\WritePrecision::MS, $bucket, $org);
```
Comment (bednar) [inline examples/RecordRowExample.php:32]: Curly braces are not required for string interpolation:

```suggestion
    $writeApi->write("point,table=my-table result=$i", InfluxDB2\Model\WritePrecision::MS, $bucket, $org);
```

File: examples/RecordRowExample.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+$writeApi->close();
+
+//
+// Query data with pivot
+//
+$queryApi = $client->createQueryApi();
+
+$result = $queryApi->query(
+    "from(bucket: \"{$bucket}\") |> range(start: -1m) |> filter(fn: (r) => (r[\"_measurement\"] == \"point\"))
```
Comment (bednar) [inline examples/RecordRowExample.php:43]: Curly braces are not required for string interpolation:

```suggestion
    "from(bucket: \"$bucket\") |> range(start: -1m) |> filter(fn: (r) => (r[\"_measurement\"] == \"point\"))
```

File: examples/RecordRowExample.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+);
+
+//
+// Write data to output
+//
+printf("\n--------------------------------------- FluxRecord.values ----------------------------------------\n");
+foreach ($result as $table) {
+    foreach ($table->records as $record) {
+        $values = implode(", ", $record->values);
+        print "{$values}\n";
```
Comment (bednar) [inline examples/RecordRowExample.php:54]: Curly braces are not required for string interpolation:
```suggestion
        print "$values\n";
```

File: examples/RecordRowExample.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+}
+
+printf("\n----------------------------------------- FluxRecord.row -----------------------------------------\n");
+foreach ($result as $table) {
+    foreach ($table->records as $record) {
+        $row = implode(", ", $record->row);
+        print "{$row}\n";
```
Comment (bednar) [inline examples/RecordRowExample.php:62]: Curly braces are not required for string interpolation:

```suggestion
        print "$row\n";
```

**Review by bednar — CHANGES_REQUESTED** [review]
Thanks for your PR 👍 
There are a few requirements that must be be satisfy before we accept the PR:

File: src/InfluxDB2/FluxCsvParser.php [hunk 5]
```diff
@@ -183,10 +185,24 @@ private function addDefaultEmptyValues(FluxTable $table, $defaultValues)
...
+
+        foreach ($table->columns as $column) {
             $column->label = $csv[$i];
+            if (in_array($column->label, $labels)) {
+                $duplicates[] = $column->label;
+            } else {
+                $labels[] = $column->label;
+            }
             $i++;
         }
```
Comment (bednar) [inline src/InfluxDB2/FluxCsvParser.php:199]: What do you think about something more easier like:

```suggestion
        $duplicates = array();

        foreach (array_count_values($csv) as $label => $count) {
            if ($count > 1) {
                $duplicates[] = $label;
            }
            $table->columns[$i]->label = $label;
            $i++;
        }
```

?

**Review by bednar — CHANGES_REQUESTED** [review]

File: src/InfluxDB2/FluxCsvParser.php [hunk 6]
```diff
@@ -183,10 +185,24 @@ private function addDefaultEmptyValues(FluxTable $table, $defaultValues)
...
+        foreach (array_count_values($csv) as $label => $count) {
+            if ($count > 1) {
+                $duplicates[] = $label;
+            }
+        }
+
+        if (count($duplicates) > 0) {
+            $duplicatesStr = implode(", ", $duplicates);
+            print "The response contains columns with duplicated names: {$duplicatesStr}\n";
+            print "You should use the 'FluxRecord.row to access your data instead of 'FluxRecord.values'.";
```
Comment (bednar) [inline src/InfluxDB2/FluxCsvParser.php:204]: Just last typo:

```suggestion
            print "You should use the 'FluxRecord.row' to access your data instead of 'FluxRecord.values'.";
```

**Review by bednar — CHANGES_REQUESTED** [review]

**Review by bednar — APPROVED** [review]
LGTM 🌹
